### PR TITLE
PR for all outputs.

### PR DESCRIPTION
**Pull request by fmustapha**
#### What does this PR do?
Merges changes in bayo-kakashi-fmaustapha to  bayo-kakashi-fmaustapha-output.
- Output for setting up continuous integration.
- Output for automating tasks.
- Output for writing tests.
- Output for delivering your product.
- Output for getting peer reviews.
- Output for solving the problem.
- Output for reflecting on the learning experience
- Output for writing HTML,JS, CSS 
- Output for checking repository health 
- Output for setting up software development environment.
- Output for setting up your repository.
- Output for using design patterns
#### Description of Task to be completed?
- Deliver a link to a successful TravisCI build of your `development` branch.
- Deliver a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files.
- Deliver test files for the Inverted Index class and all associated dependencies.
- Deliver `Release notes` for your final product in your repo's wiki
- Deliver a Pull Request showing at least two Peer Code Reviews
- Deliver the file containing the implementation including all relevant dependencies. (e.g. InvertedIndex class).
- Deliver a post reflecting on your learning experience while producing these outputs.
- Deliver a folder containing the template for the proposed frontEnd of your project.
- Deliver a screenshot of your code coverage and GPA badges in your repo's README file.
- Deliver live URLs to a review app, your staging & production apps.
- Deliver the repository for your proposed project and at least one `Pull Request`.
- Deliver a Pull Request showing two Design Patterns (suggestions: Decorator, Factory, Observer, Prototype)

#### How should this be manually tested?
- For setting up continuous integration output click here [View](https://travis-ci.org/andela-fmustapha/inverted-index/builds/217587387)
- For automating tasks output click here [View](https://github.com/andela/test-simulations-mgmt-repo/blob/bayo-kakashi-fmustapha/06-building-a-program-start-to-finish/automating-tasks/output/README.md)
- For writing tests output click here [View](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-fmustapha/06-building-a-program-start-to-finish/writing-tests/output)
- For delivering your product output click here [View](https://docs.google.com/document/d/1W-MOn4Uav1vILaws19-eJ2K-Y11VNv_2NYEhhi45htA/edit).
- For getting peer reviews output click here [View](https://github.com/andela-fmustapha/inverted-index/pull/12).
- For solving the problem output click here [View](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-fmustapha/06-building-a-program-start-to-finish/solving-the-problem/output). 
- For reflecting on the learning experience output click here [View](https://medium.com/@fyoiza/reflecting-on-my-learning-experience-in-andela-so-far-2476013ceefb)
- For writing HTML,JS, CSS output click here [view](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-fmustapha/06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output).
- For checking repository health output click here [view](http://i.imgur.com/Z5dccQx.png)
- For setting up software development environment click here [view](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-fmustapha/06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output).
- For setting up your repository, view my repository here [view](https://github.com/andela-fmustapha/inverted-index).
- For using design patterns, view output here [View](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-fmustapha/06-building-a-program-start-to-finish/using-design-patterns/output)

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)
Output for getting-peer-reviews 
screenshot 2017-04-03 00 30 20

#### Questions:
N/A

### CODE REVIEW

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [Output for Gulp automation](https://www.youtube.com/watch?v=7ffZj81sxnc)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/output/README.md]: This shows for only your HTML and CSS... What about the JS?

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## Checking Repository Health
+----
+#### Output
+
+- **Using**: `Code Climate` and `Coveralls`
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+
+##### README 
+- Readme screenshot
+- ![badge screenshot](http://i.imgur.com/Z5dccQx.png)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:10]: I see your badges. Cool

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: a Pull Request showing at least two Peer Code Reviews
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [View PR](https://github.com/andela-fmustapha/inverted-index/pull/12)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:7]: Cool. Hope you were able to implemented the feedback you received.

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Medium
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [Reflecting on the learning experience](https://medium.com/@fyoiza/reflecting-on-my-learning-experience-in-andela-so-far-2476013ceefb)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md:7]: Good read.

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: TravisCI
+- **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
+- **With**: *independently*
+
+#### Link to a successful TravisCI build of my development branch
+- [Travis on Development](https://travis-ci.org/andela-fmustapha/inverted-index/builds/217587387)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:7]: Cool

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#### Output
+- **Using**: Heroku
+- **Deliver**: live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [Staging](https://book-switch-staging.herokuapp.com/)
+- [Review](https://book-switch-production-pr-12.herokuapp.com/)
+- [Production](https://book-switch-production.herokuapp.com/)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md]: You have an error on production. Any reason why?

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: the repository for your proposed project and at least one `Pull Request`
+- **With**: *independently*
+
+### Link to GitHub Repo
+
+- [My InvertedIndex Repo](https://github.com/andela-fmustapha/inverted-index/ "Visit Faith's Project Repo")
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:8]: Cool

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js [hunk 8]
```diff
@@ -0,0 +1,209 @@
+/**
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: This file name is not proper. Please meet with your peers on what is preferred.

Also review your parameter/variable names in this entire file. 

You multiple cases of eslint "_**exceeds the maximum line length of 80**_" violations on this file. Please them all

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product in your repo's wiki
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [Release Document v1.0.0](https://docs.google.com/document/d/1W-MOn4Uav1vILaws19-eJ2K-Y11VNv_2NYEhhi45htA/edit "BookSwitch Release Note")
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:7]: Please provide access.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+    const indices = {};
+    const splittedWords = {};
+    book.forEach((doc, key) => {
+      const joinedkeys = InvertedIndex.concatenateText(doc);
+      const tokenizedWords = InvertedIndex.tokenizeWords(joinedkeys);
+      splittedWords[key] = InvertedIndex.splitAndSort(tokenizedWords);
+    });
+    Object.keys(splittedWords).forEach((keys) => {
+      splittedWords[keys].forEach((words) => {
+        if (!indices.hasOwnProperty(words)) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: I believe there is a better way to call `Object.prototype` methods directly. Please [see link](https://github.com/airbnb/javascript#objects--prototype-builtins)

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+
+  /**
+   *
+   *
+   * @param {Object} eachFile - Javascript object containing file properties
+   * @returns {Object} fileContent - content of the read file
+   *
+   * @memberOf InvertedIndex
+   */
+  readFile(eachFile) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: Did you get an _eslint_ (_Expected 'this' to be used by class method_) error here?

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+  it('should return false if file is not an array of JSON object',
+     () => {
+       expect(InvertedIndex.validateFile(invalidBook)).toBeFalsy();
+     });
+
+  it('should return false if file contains an empty array',
+     () => {
+       const msg = { status: false };
+       expect(InvertedIndex.validateFile(malformedJSON)).toEqual(msg);
+     });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: From line 45 to line 84should be in a separate describe block.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+  describe('Search index', () => {
+    it('should return true if search term is a string', () => {
+      const term = 'Wonderland of rings';
+      expect(Object.keys(invertedIndex.searchIndex(term, 'smallValidBook.json')))
+      .toBeTruthy();
+    });
+  });
+
+  it('should search through single files that are indexed', () => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Why it this test spec not in a test suite?

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+        unusual: [1, 2]
+      }
+    };
+    let searchTerm = {};
+    searchTerm = invertedIndex.searchIndex('Alice, and her unusual imagination',
+      'smallValidBook.json');
+    expect(Object.keys(searchTerm)).toEqual(Object.keys(requiredOutput));
+    expect(searchTerm).toEqual(requiredOutput);
+  });
+});
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: I do not see test suite for some off your methods? Are reason why you didn't add them?

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js [hunk 15]
```diff
@@ -0,0 +1,152 @@
+const validBook = require('./test/testFiles/validBook.json');  //  book with valid contents
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: You have multiple cases of eslint "_**exceeds the maximum line length of 80**_" violations in this test file. Please fix them all

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+const invalidBook = require('./test/testFiles/invalidBook.json'); //  book with invalid content
+const emptyBook = require('./test/testFiles/emptyBook.json');   //  empty book
+const malformedJSON = require('./test/testFiles/malformed.json'); // empty book
+const smallBook = require('./test/testFiles/smallValidBook.json'); // book with few words
+const InvertedIndex = require('../src/js/inverted.js').InvertedIndex; // Inverted Index class file
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: A base filename should exactly match the name of its default export. Your filenames and variable names should be in camelCase. It aids readability.

**Review by andela-moe — COMMENTED** [review]
Please address comments.

**Comment by andela-moe** [comment]
Please address comments and re-open when done. Thanks

File: 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md (hunk 3 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/getting-peer-reviews/output/README.md:7]: Yes I was able to, thank you.

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md (hunk 4 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/output/README.md:7]: Thank you.

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md (hunk 5 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/output/README.md:7]: Thank you.

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md (hunk 6 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md]: I will check and make necessary corrections.

File: 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md (hunk 7 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/setting-up-your-repository/output/README.md:8]: Thank you.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js (hunk 8 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: Okay thank you.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js (hunk 11 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: I don't remember getting that error, but I will check and make necessary changes.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js (hunk 10 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: Okay thank you.

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md (hunk 9 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:7]: Okay thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 15 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Okay thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 16 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Okay thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 12 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Okay thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 13 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: I will fix that, thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 14 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: I will include them, thank you.

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md (hunk 1 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/automating-tasks/output/README.md]: I have now done another video that covers HTML, CSS and JS automation using Gulp.

File: 06-building-a-program-start-to-finish/checking-repository-health/output/README.md (hunk 2 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/checking-repository-health/output/README.md:10]: Thank you.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js (hunk 11 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: I have now added static to readFile method and removed "class-methods-use-this": 0,
from my .eslintrc.json file.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js (hunk 10 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted.js]: I have now removed  "no-prototype-builtins": 0, from my .eslintrc.json file and used 'Object.indices.hasOwnProperty.call(indices, words) '

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md (hunk 9 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:7]: Access has now been provided. Thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 15 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Fixed, thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 16 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Fixed, thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 12 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Fixed, thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 13 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: Fixed, thank you.

File: 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js (hunk 14 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/invertedSpec.js]: I have now included test suites for all but one of my methods (i.e. 'readFile' method).
For the readFile method, I used HTML5 fileReader API which currently cannot be tested because it requires an actual file upload.

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md [hunk 17]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [Output for Gulp automation](https://youtu.be/k7yfGIM2ynQ)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/output/README.md:7]: Nice

File: 06-building-a-program-start-to-finish/delivering-your-product/output/README.md (hunk 9 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/delivering-your-product/output/README.md:7]: Nice

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md [hunk 18]
```diff
@@ -0,0 +1,9 @@
+#### Output
+- **Using**: Heroku
+- **Deliver**: live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+- [Staging](https://book-switch-staging.herokuapp.com/)
+- [Review](https://book-switch-staging-pr-12.herokuapp.com/)
+- [Production](https://book-switch-production.herokuapp.com/)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:9]: Cool. Live urls

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js [hunk 19]
```diff
@@ -0,0 +1,237 @@
+/**
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: I like you attempt at modifying the name of this file. This is snake_case not what is expected for this file. Please confirm from your peers.

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+  createIndex(book, filename) {
+
+  }
+
+```
+To use the methods of the object you first have to create an instance of the class by doing: 
+`invertedIndex = new InvertedIndex()`
+
+#### AUTHOR
+- **Faith Mustapha**
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:41]: Good

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/index.html [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+    <div id="footer">Copyright &copy; FAM.com</div>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+    <script src="https://code.angularjs.org/1.3.0-rc.1/angular.min.js"></script>
+    <script src="js/inverted-index.js"></script>
+    <script src="js/indexController.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/toastr.js/latest/js/toastr.min.js"></script>
+    </div>
+</body>
+
+</html>
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/index.html]: Good attempt with your UI. Your controls look a little too lean. I Feel you could have leveraged on a css framework to help you there.

**Review by andela-moe — COMMENTED** [review]
Please review. Address comments

File: 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js (hunk 19 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js]: I have now modified the file name.

File: 06-building-a-program-start-to-finish/using-design-patterns/output/README.md (hunk 20 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/using-design-patterns/output/README.md:41]: Thank you.

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/index.html (hunk 21 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/index.html]: Yes, I am now using bootstrap for my sims tasks and I'll do same for my CP 2.

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md (hunk 18 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/output/README.md:9]: Thank you.

File: 06-building-a-program-start-to-finish/automating-tasks/output/README.md (hunk 17 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/automating-tasks/output/README.md:7]: Thanks.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js [hunk 22]
```diff
@@ -0,0 +1,249 @@
...
+    it('should check that the contents of the uploaded file is valid',
+    () => {
+      const msg = { status: true,
+        jsonContent:
+        [{ title: 'Alice in Wonderland',
+          text: 'Alice falls into a rabbit hole and enters a world '
+          + 'full of imagination.' },
+        { title: 'The Lord of the Rings: The Fellowship of the Ring.',
+          text: 'An unusual alliance of man, elf, dwarf, wizard and hobbit '
+          + 'seek to destroy a powerful ring.' },
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js]: Please make use of template literals instead of concatenating the string this way OR is it possible to have this array objects in file

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js [hunk 23]
```diff
@@ -0,0 +1,249 @@
...
+        { title: 'The Lord of the Rings: The Fellowship of.',
+          text: 'An unusual alliance of man, elf, dwarf, wizard and hobbit '
+          + 'seek to destroy a powerful ring.' }] };
+      expect(InvertedIndex.validateFile(validBook)).toEqual(msg);
+    });
+
+    it('should return false for empty json files', () => {
+      const successMsg = { status: false };
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js]: what is this line space for?

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/js/InvertedIndex.js [hunk 24]
```diff
@@ -0,0 +1,236 @@
...
+    let isValid = { status: false };
+    const invalidStructureError = {
+      name: 'validate file structure',
+      message: 'File structure is invalid'
+    };
+    if (Object.keys(jsonContent).length === 0 &&
+       typeof jsonContent === 'object') {
+      isValid = {
+        status: false
+      };
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/js/InvertedIndex.js]: do you need to do this here again? _**isValid**_ at this point has not changed from what it is on line 50

**Review by andela-moe — COMMENTED** [review]
Please review comments

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js [hunk 25]
```diff
@@ -0,0 +1,233 @@
...
+   *  returns the concatenated values of object keys
+   *
+   * @memberOf InvertedIndex
+   */
+  static concatenateText(document) {
+    // if a document exists? combine title and text to split at once
+    let concatenatedText = '';
+    concatenatedText = `${document.title} ${document.text}`;
+    return concatenatedText;
+  }
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: I believe this method `concatenateText()` is really not necessary. Consider moving line 143 to your `createIndex()`

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js [hunk 26]
```diff
@@ -0,0 +1,255 @@
+const validBook = require('./test/testFiles/validBook.json');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js]: This `validBook` is defined but not used. Do you need to still keep it here?

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js [hunk 27]
```diff
@@ -0,0 +1,255 @@
...
+    it('should return false if file contains an empty array',
+     () => {
+       const msg = { status: false };
+       expect(InvertedIndex.validateFile(malformedBook)).toEqual(msg);
+     });
+  });
+
+  describe('The createIndex method', () => {
+    it('should return mapped indices to words in a JSON file', () => {
+      invertedIndex.createIndex(validBook2, 'validBook2.json');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js]: Why is this line here?

**Review by andela-moe — COMMENTED** [review]
Please review comments

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js (hunk 25 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js]: Updated.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js (hunk 26 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js]: Removed book.

File: 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js (hunk 27 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js]: Test has been updated.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js [hunk 28]
```diff
@@ -0,0 +1,216 @@
...
+          };
+          return isValid;
+        }
+      });
+    } catch (error) {
+      if (error.name === 'validate file structure') return isValid;
+      throw error;
+    }
+    return isValid;
+  }
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js:76]: I tried your [application](https://book-switch-staging.herokuapp.com/) and It appears you are not validating that the content is properly formatted. Please fix that and also provide tests for that too.

Check your browsers console for these errors

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/js/indexController.js [hunk 29]
```diff
@@ -0,0 +1,99 @@
...
+}]);
+
+app.factory('alertFactory', () => ({
+  success: (text) => {
+    toastr.success(text, 'Success');
+  },
+  error: (text) => {
+    toastr.error(text, 'Error');
+  }
+}));
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/js/indexController.js:117]: I think you should try uploading a file with a large number of documents with similar titles. I uploaded a file and the titles didn't display.

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/js/indexController.js (hunk 29 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output/src/js/indexController.js:117]: Okay, I have fixed this.

File: 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js (hunk 28 above)
Comment (fmustapha) [reply 06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js:76]: I have fixed this also, but I am not able to write a test for this because, I cannot require a file with invalid JSON file format.
